stat.c

```diff
@@ -3087,17 +3091,30 @@ int calc_log_samples(void)
 		if (!td->bw_log ||
 			(td->bw_log && !per_unit_log(td->bw_log))) {
 			tmp = add_bw_samples(td, &now);
-			if (tmp < next)
-				next = tmp;
+
+			if (td->bw_log)
+				log_avg_msec = td->bw_log->avg_msec;
 		}
 		if (!td->iops_log ||
 			(td->iops_log && !per_unit_log(td->iops_log))) {
 			tmp = add_iops_samples(td, &now);
-			if (tmp < next)
-				next = tmp;
+
+			if (td->iops_log)
+				log_avg_msec = td->iops_log->avg_msec;
 		}
+
+		if (tmp < next)
+			next = tmp;
 	}
 
+	if (log_avg_msec == 0)
```

Comment:
I'm have some questions here:
- Can different threads have different log_avg_msec values? If so does it mean we need the lowest one that is greater than 0?
- Could we just do 
 `max(td->o.iops_avg_time, td->o.bw_avg_time)` and check if the result were 0 to see if either one had been set within a loop iteration?

Reply:
I'm not sure if different threads can have different log_avg_msec values.  @vincentkfu or @bvanassche  do you know the answer to this?

Reply:
It's a per-job option, so safe to assume that different threads can have different values.

Reply:
@jeffreyalien Doesn't `log_avg_msec` only account for the "last" thread with it set?

Reply:
@sitsofe Yes, you're correct about the log_avg_msec being from the last thread.  I could add code track and use the minimum log_avg_msec  of all the threads.   I believe that should solve the potential issue of having threads with different values.

Reply:
@jeffreyalien sounds good!